### PR TITLE
Switch to intra-doc links in `std::process`

### PR DESCRIPTION
**Pull request by camelid**
Part of #75080.

### CODE REVIEW

File: library/std/src/process.rs [hunk 1]
```diff
@@ -227,9 +215,8 @@ impl fmt::Debug for Child {
 /// file handle will be closed. If the child process was blocked on input prior
 /// to being dropped, it will become unblocked after dropping.
 ///
-/// [`Child`]: struct.Child.html
-/// [`stdin`]: struct.Child.html#structfield.stdin
-/// [dropped]: ../ops/trait.Drop.html
+/// [`stdin`]: crate::process::Child.stdin
```
Comment (camelid) [inline library/std/src/process.rs:218]: This part isn't working. I tried `Child.stdin` too, but that didn't work either. What should I do to get it to work?

File: library/std/src/process.rs [hunk 2]
```diff
@@ -286,9 +273,8 @@ impl fmt::Debug for ChildStdin {
 /// When an instance of `ChildStdout` is [dropped], the `ChildStdout`'s
 /// underlying file handle will be closed.
 ///
-/// [`Child`]: struct.Child.html
-/// [`stdout`]: struct.Child.html#structfield.stdout
-/// [dropped]: ../ops/trait.Drop.html
+/// [`stdout`]: crate::process::Child.stdout
```
Comment (camelid) [inline library/std/src/process.rs:276]: Same as https://github.com/rust-lang/rust/pull/75432/files#r468959360

File: library/std/src/process.rs [hunk 3]
```diff
@@ -347,9 +333,8 @@ impl fmt::Debug for ChildStdout {
 /// When an instance of `ChildStderr` is [dropped], the `ChildStderr`'s
 /// underlying file handle will be closed.
 ///
-/// [`Child`]: struct.Child.html
-/// [`stderr`]: struct.Child.html#structfield.stderr
-/// [dropped]: ../ops/trait.Drop.html
+/// [`stderr`]: crate::process::Child.stderr
```
Comment (camelid) [inline library/std/src/process.rs:336]: Same as https://github.com/rust-lang/rust/pull/75432/files#r468959360

File: library/std/src/process.rs (hunk 1 above)
Comment (jyn514) [reply library/std/src/process.rs:218]: Try `Child::stdin` instead. Looks like this isn't documented on https://doc.rust-lang.org/nightly/rustdoc/unstable-features.html#linking-to-items-by-name, @Manishearth are you interested in fixing that?

As a side note the syntax for this is a little confusing ... maybe we should support both `::` and `.` for fields?

Comment (camelid) [reply library/std/src/process.rs:218]: Hmm, that's weird – [the RFC says](https://github.com/rust-lang/rfcs/blob/68e17ace829e81239fe5020b7a6dbc22550b6d16/text/1946-intra-rustdoc-links.md#linking-to-fields) to use `Struct.field` to reference a struct field:

> ## Linking to Fields
>
> To link to the fields of a struct we propose to write the path to the struct, followed by a dot, followed by the field name.
>
> For example:
>
> ```markdown
> This is stored in the [`size`](storage::Filesystem.size) field.
> ```

Comment (jyn514) [reply library/std/src/process.rs]: Yeah the implementation is out of sync with the RFC in some places, I guess this is one of them. I'll open an issue.

Comment (jyn514) [reply library/std/src/process.rs]: Opened https://github.com/rust-lang/rust/issues/75437.

**Comment by camelid** [comment]
@jyn514 Do you want me to ask @KodrAus to review this since rust-highfive assigned him at first?

**Comment by jyn514** [comment]
Whoops, missed that you addressed the comments.

@bors r+ rollup

**Comment by camelid** [comment]
That's okay! Thanks for reviewing this!

**Pull request merged**